plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaInternalFieldManager.java

```diff
@@ -134,59 +156,76 @@ ColumnMetadata getColumnMetadata(boolean hidden)
     private final Map<String, InternalField> internalFields;
 
     @Inject
-    public KafkaInternalFieldManager(TypeManager typeManager)
+    public KafkaInternalFieldManager(TypeManager typeManager, KafkaConfig kafkaConfig)
     {
         Type varcharMapType = typeManager.getType(mapType(VARCHAR.getTypeSignature(), arrayType(VARBINARY.getTypeSignature())));
-
-        internalFields = ImmutableMap.<String, InternalField>builder()
-                .put(PARTITION_ID_FIELD, new InternalField(
-                        PARTITION_ID_FIELD,
-                        "Partition Id",
-                        BigintType.BIGINT))
-                .put(PARTITION_OFFSET_FIELD, new InternalField(
-                        PARTITION_OFFSET_FIELD,
-                        "Offset for the message within the partition",
-                        BigintType.BIGINT))
-                .put(MESSAGE_CORRUPT_FIELD, new InternalField(
-                        MESSAGE_CORRUPT_FIELD,
-                        "Message data is corrupt",
-                        BooleanType.BOOLEAN))
-                .put(MESSAGE_FIELD, new InternalField(
-                        MESSAGE_FIELD,
-                        "Message text",
-                        createUnboundedVarcharType()))
-                .put(HEADERS_FIELD, new InternalField(
-                        HEADERS_FIELD,
-                        "Headers of the message as map",
-                        varcharMapType))
-                .put(MESSAGE_LENGTH_FIELD, new InternalField(
-                        MESSAGE_LENGTH_FIELD,
-                        "Total number of message bytes",
-                        BigintType.BIGINT))
-                .put(KEY_CORRUPT_FIELD, new InternalField(
-                        KEY_CORRUPT_FIELD,
-                        "Key data is corrupt",
-                        BooleanType.BOOLEAN))
-                .put(KEY_FIELD, new InternalField(
-                        KEY_FIELD,
-                        "Key text",
-                        createUnboundedVarcharType()))
-                .put(KEY_LENGTH_FIELD, new InternalField(
-                        KEY_LENGTH_FIELD,
-                        "Total number of key bytes",
-                        BigintType.BIGINT))
-                .put(OFFSET_TIMESTAMP_FIELD, new InternalField(
-                        OFFSET_TIMESTAMP_FIELD,
-                        "Message timestamp",
-                        TIMESTAMP_MILLIS))
-                .buildOrThrow();
+        String prefix = kafkaConfig.getInternalFieldPrefix();
+        internalFields = Stream.of(
```

Comment:
we have this to preserve order of columns, correct?

Earlier code the order being preserved was mostly an impl detail of ImmutableMap that insertion order = iteration order?

Reply:
Don't know about ordering. I guess it is not relevant here.

This code is tricky. 
Once it looked like enum already. And was refactored this way to use dynamic value from `TypeManager`.